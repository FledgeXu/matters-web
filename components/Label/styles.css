.label {
  color: var(--color-white);
  background: var(--color-matters-green);
  user-select: none;
}

.small {
  padding: var(--spacing-xxx-tight) var(--spacing-x-tight);
  font-size: var(--font-size-xs);
}

.default {
  @mixin font-serif;

  padding: var(--spacing-xx-tight) var(--spacing-tight);

  font-size: var(--font-size-sm);
  font-weight: var(--font-weight-bold);
  line-height: 1.42857143;
}
