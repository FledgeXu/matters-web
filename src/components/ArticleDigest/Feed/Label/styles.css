.label {
  @mixin inline-flex-center-all;

  padding: var(--spacing-xxx-tight) var(--spacing-x-tight);
  font-size: var(--font-size-xs);
  line-height: 1rem;
  color: var(--color-grey);
  border: 1px solid var(--color-grey-light);
  border-radius: var(--spacing-loose);
}
