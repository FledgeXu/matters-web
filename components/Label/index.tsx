import classNames from 'classnames'

import styles from './styles.css'

type LabelSize = 'small' | 'default'

interface LabelProps {
  text: string
  size?: LabelSize
  [key: string]: any
}

/**
 *
 * Usage:
 *
 * ```tsx *
 * <Label size="small" text="作者推薦" />
 * ```
 */

export const Label: React.SFC<LabelProps> = ({
  text,

  size = 'default',

  className,
  ...restProps
}) => {
  const labelClasses = classNames({
    label: true,
    [size]: true,
    [className]: !!className
  })

  return (
    <>
      <span className={labelClasses} {...restProps}>
        {text}
      </span>
      <style jsx>{styles}</style>
    </>
  )
}
