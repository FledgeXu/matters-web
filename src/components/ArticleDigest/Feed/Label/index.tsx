import styles from './styles.css'

const Label: React.FC = ({ children }) => (
  <span className="label">
    {children}

    <style jsx>{styles}</style>
  </span>
)

export default Label
