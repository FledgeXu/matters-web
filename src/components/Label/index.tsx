import classNames from 'classnames'

import styles from './styles.css'

type LabelSize = 'small' | 'default'

interface LabelProps {
  size?: LabelSize
  style?: React.CSSProperties
}

/**
 *
 * Usage:
 *
 * ```tsx *
 * <Label size="small">作者推薦</Label>
 * ```
 */

export const Label: React.FC<LabelProps> = ({
  size = 'default',
  children,
  style
}) => {
  const labelClasses = classNames({
    label: true,
    [size]: true
  })

  return (
    <span className={labelClasses} style={style}>
      {children}

      <style jsx>{styles}</style>
    </span>
  )
}
