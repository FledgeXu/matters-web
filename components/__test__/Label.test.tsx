import React from 'react'
import { render } from 'react-testing-library'

import { Label } from '../Label'

test('Label default style match snapshot', () => {
  const { container } = render(<Label />)
  expect(container.firstChild).toMatchSnapshot()
})
