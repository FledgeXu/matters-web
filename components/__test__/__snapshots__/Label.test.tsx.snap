// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label default style match snapshot 1`] = `
<span
  class="jsx-undefined label default"
/>
`;
